### ISSUE TITLE
Discussion: The portable PDB support

### ISSUE THREAD

**Issue by ww898** [dotnet/coreclr#8175]
Hi,
I would to open the discussion about portable PDB support in CoreCLR. Now our product JetBrains profiler is using the COM object from `diasymreader.dll` (interfaces `ISymUnmanagedReader` / `ISymUnmanagedBinder2`) for windows PDB format (`Microsoft C/C++ MSF 7.00`). Now it possible to select what type of PDB will be generated for CoreCLR project due `debugType` option in `project.json` : `full` (windows specific) or `portable` (metadata based with signatures `BSJB` / `MPDB`).

I spent last two weeks for writing my own C++11 library (with small piece of boost v1.62.0) for parsing the PE-file, the metadata `BSJB`, and deflating the `MPDB` blob. I'm finished it yesterday. Everything works, but I still have questions:

1. What should I do when user started to profile .NET executable with windows specific PDB on Linux or MacOS? There is no COM object from `diasymreader.dll` on these OS. However, I found `Microsoft.DiaSymReader.Native.*.dll`. After investigation I assume that this is the libraries for windows specific PDB format support. Am I right? What kind of license they have? Why is there no source on GitHub for them? Can we redistribute these libraries with our products?
2. Do you plans to implement a native version of library for support portable PDB on MSVC/GCC/Clang C/C++ compilers?
3. What with PDB server access? I mean `srv*`, `symsrv*`, `cache*` paths. The `Microsoft.DiaSymReader.PortablePdb.dll` is also not implement this feature yet.

**Comment by jkotas** [comment]
cc @tmat

**Comment by tmat** [comment]
1. Windows PDBs are not supported on Linux/MacOS. You are right, ```Microsoft.DiaSymReader.Native.*.dll``` packages a standalone build of DiaSymReader that provides DLL entry points and doesn't require COM registration. It's used by and distributed with Roslyn compilers and dotnet cli. DiaSymReader is not open source.
2. We don't currently have plans to implement a native version of Portable PDB reader/writer. The managed implementation is provided by System.Reflection.Metadata package.
3. symsrv currently only supports Windows PDBs and PE files. No other formats are supported by the symbol server. We are considering various options for symbol server support at this moment. You can follow https://github.com/dotnet/symstore for further development.

**Comment by tmat** [comment]
cc @noahfalk

**Comment by ww898** [comment]
Does it mean that .NET executables which was compiled with Windows PDB can properly work on Linux / MacOS, but, at the same time, a CoreCLR profiler can't instrument IL code, because there is no official way to excavate per statement IL offsets from Windows PDB?

**Comment by tmat** [comment]
Yes, if the profiler needs to read line info from a Windows PDB it won't work on Linux/MacOS. Portable PDBs were introduced to support x-plat debugging/tooling.

**Comment by pkruk2** [comment]
Hello, 

According to https://github.com/Microsoft/vstest/issues/800#issuecomment-373310602 DIA SDK has been updated to support Portable PDB format. It is available with VS 15.8 preview 3. I have verified it works (on one .pdb file, but still, it didn't work previously).

Does this mean that there is a chance for support for Portable PDB in Profiling API?

(I'm not sure if it's possible to use DIA SDK in Profiling API due to licensing reasons - the SDK is distributed with VS and Build Tools and I believe it requires VS license - although it's not specified explicitly and I've seen contradicting statements on this topic).
